tests/acceptance/common.sh

```diff
@@ -0,0 +1,138 @@
+function log() {
+    local level="${2:-INFO}"
+    echo "[$(TZ=UTC date --rfc-3339=ns) ${level} ${BATS_TEST_NAME}] $1"
+}
+
+function logError() {
+    log $1 'ERROR'
+}
+
+function logWarn() {
+    log $1 'WARN'
+}
+
+function testAzureSecret() {
+    local role_name="$1"
+    local subscription_id="$2"
+    local resource_group_name="$3"
+    local vault_role_name="$4"
+    local config_dir="$5"
+    local engine_name="$6"
+    local ttl=5
+
+    log "Creating Azure secret, azure_role='${role_name}', vault_role='${vault_role_name}'"
+    vault write "${engine_name}/roles/${vault_role_name}" ttl="${ttl}" azure_roles=-<<EOF
+[
+    {
+        "role_name": "${role_name}",
+        "scope":  "/subscriptions/${subscription_id}/resourceGroups/${resource_group_name}"
+    }
+]
+EOF
+
+    local secret_file="${config_dir}/secret-${vault_role_name}.json"
+    vault read "${engine_name}/creds/${vault_role_name}" -format=json > ${secret_file}
+
+    sp_id=$(cat "${secret_file}" | jq -er .data.client_id)
+    log "Secret created successfully, azure_role='${role_name}', vault_role='${vault_role_name}', file=${secret_file}"
+
+    if !assertSPExistence ; then
+        exit 1
```

Comment:
Was a little surprised this is `exit` instead of `return`, but is that because each `@test` is running in a subshell?

Reply:
Fixed in https://github.com/hashicorp/vault-plugin-secrets-azure/pull/85/commits/de8335c6c878e2d9b19e4dcfaaefb6038dd6aa29